jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildImageMojo.java

```diff
@@ -89,13 +90,20 @@ public void execute() throws MojoExecutionException, MojoFailureException {
           "Authentication over HTTP is enabled. It is strongly recommended that you do not enable "
               + "this on a public network!");
     }
+
     MavenSettingsServerCredentials mavenSettingsServerCredentials =
         new MavenSettingsServerCredentials(
             Preconditions.checkNotNull(session).getSettings(), settingsDecrypter, mavenBuildLogger);
+    Authorization fromAuthorization = getBaseImageAuth();
     RegistryCredentials knownBaseRegistryCredentials =
-        mavenSettingsServerCredentials.retrieve(baseImage.getRegistry());
+        fromAuthorization != null
+            ? new RegistryCredentials("<jib><from><auth>", fromAuthorization)
+            : mavenSettingsServerCredentials.retrieve(baseImage.getRegistry());
+    Authorization toAuthorization = getTargetImageAuth();
     RegistryCredentials knownTargetRegistryCredentials =
-        mavenSettingsServerCredentials.retrieve(targetImage.getRegistry());
+        toAuthorization != null
+            ? new RegistryCredentials("<jib><to><auth>", toAuthorization)
+            : mavenSettingsServerCredentials.retrieve(targetImage.getRegistry());
```

Comment:
I wonder if this code could be pushed up to `JibPluginConfiguration` and added by providing the `BuildConfiguration.Builder` instance. For example
```
  setBaseImageCredentialProvisioning(baseConfigurationBuilder);
  setTargetImageCredentialProvisioning(baseConfigurationBuilder);
```

It would reduce code duplicated across the mojos.

Reply:
I think in general we're trying to keep `JibPluginConfiguration` as a class that just contains all the Maven configuration properties without much extra logic. If anything I should probably be moving the system property/build config choosing stuff out of there, but I left it since I'm essentially using the system properties as a replacement for commandline properties (because I couldn't figure out how to tag the correct auth parameters as such, and system properties are easier and identical as far as setting them is concerned). We can work on reducing duplicate code when we work on the `jib-plugins-common` module.